src/Sentry.AspNetCore.Grpc/SentryGrpcInterceptor.cs

```diff
@@ -0,0 +1,236 @@
+        {
+            var hub = _hubAccessor();
+            if (!hub.IsEnabled)
+            {
+                return await continuation(requestStream, context).ConfigureAwait(false);
+            }
+
+            using (hub.PushAndLockScope())
+            {
+                hub.ConfigureScope(scope =>
+                {
+                    scope.OnEvaluating += (_, __) => scope.Populate<TRequest>(context, null, _options);
+                });
+
+                try
+                {
+                    return await continuation(requestStream, context).ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    CaptureException(hub, e);
+
+                    ExceptionDispatchInfo.Capture(e).Throw();
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Handles the <see cref="Interceptor.ServerStreamingServerHandler{TRequest,TResponse}"/> while capturing any errors
+        /// </summary>
+        /// <param name="requestStream">The request stream</param>
+        /// <param name="responseStream">The response stream</param>
+        /// <param name="context">The server call context</param>
+        /// <param name="continuation">The continuation</param>
+        /// <typeparam name="TRequest">The request type</typeparam>
+        /// <typeparam name="TResponse">The response type</typeparam>
+        /// <returns></returns>
+        public override async Task DuplexStreamingServerHandler<TRequest, TResponse>(
+            IAsyncStreamReader<TRequest> requestStream, IServerStreamWriter<TResponse> responseStream,
+            ServerCallContext context, DuplexStreamingServerMethod<TRequest, TResponse> continuation)
+        {
+            var hub = _hubAccessor();
+            if (!hub.IsEnabled)
+            {
+                await continuation(requestStream, responseStream, context).ConfigureAwait(false);
+
+                return;
+            }
+
+            using (hub.PushAndLockScope())
+            {
+                hub.ConfigureScope(scope =>
+                {
+                    scope.OnEvaluating += (_, __) => scope.Populate<TRequest>(context, null, _options);
+                });
+
+                try
+                {
+                    await continuation(requestStream, responseStream, context).ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    CaptureException(hub, e);
+
+                    ExceptionDispatchInfo.Capture(e).Throw();
+                }
+            }
+        }
+
+        private void CaptureException(IHub hub, Exception e)
+        {
+            var evt = new SentryEvent(e);
+
+            _logger.LogTrace("Sending event '{SentryEvent}' to Sentry.", evt);
+
+            var id = hub.CaptureEvent(evt);
+
+            _logger.LogInformation("Event '{id}' queued.", id);
```

Comment:
Correct me if I'm wrong but this would be the first time we use `ILogger` in the SDK directly, right? So far we've stuck to `sentryOptions.Logger?.Log...`.
In ASP.NET Core (and MEL) we hook into the frameworks logging so you can actually see these logs everywhere you pipe them through, but only if you've opt-ed in to Sentry's logging in the first place (sentryOptions.Debug=true). With this approach you took Sentry's logs will always be send to the applications logs, even if Debug=false.

I propose we stick to `options.DiagnosticLogger?.LogXyz`

Reply:
@bruno-garcia I actually took that part directly from the ASP.NET Core middleware: https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry.AspNetCore/SentryMiddleware.cs#L134

Reply:
I guess I was the first offender. Since we have presendece in the aspnet integration, looks good to keep it